### DISCUSSION TITLE
what is +page.server.js on reload?

### DISCUSSION THREAD

**Discussion by jdgamble555** [sveltejs/kit#11930] [Q&A]
After you load the route for "+page.server.js", and the route changes, does it call an RPC, does it create an endpoint, or does it just reload the whole route dynamically under the hood?

I know security is highly important, so wasn't sure...

J

**Comment by kyllerss** [comment]
You should be able to quickly test this to validate your concerns, but in my experience, it will invoke `+page.server.js` transparently under-the-hood (assuming this is not a static site).

**Comment by jdgamble555** [comment]
I figured it out. It loads just the page data by calling `YOUR/ROUTE/__data.json?x-sveltekit-invalidated=001`. This will call the route, but only return the page server load data.

You can see this in the network tab when you navigate.

J
